src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -323,16 +318,61 @@ class RideHailManager(
       val persons: Iterable[Person] = RandomUtils
         .shuffle(scenario.getPopulation.getPersons.values().asScala, rand)
         .take(numRideHailAgents.toInt)
+      val vehicleTypes = VehiclesAdjustment
+        .getVehicleAdjustment(beamServices)
+        .sampleRideHailVehicleTypes(
+          numVehicles = numRideHailAgents.toInt,
+          vehicleCategory = VehicleCategory.Car
+        )
+      var activityEndTimes: ArrayBuffer[Int] = new ArrayBuffer[Int]()
+      scenario.getPopulation.getPersons.asScala.map(
+        _._2.getSelectedPlan.getPlanElements.asScala
```

Comment:
I would suggest using a `.collect` here:

```
_._2.getSelectedPlan.getPlanElements.asScala.collect{ 
  case activity: Activity if activity.getEndTime.toInt > 0 => activity.getEndTime.toInt 
}.foreach(activityEndTimes += _)
```

At minimum note that you should use a `foreach` over a `map` since you are not creating anything - and it makes it explicit that you are mutating.